letsencrypt/cli.py

```diff
@@ -695,6 +725,168 @@ def install(config, plugins):
+    try:
+        _restore_required_config_elements(full_path, config, renewalparams)
+        _restore_plugin_configs(config, renewalparams)
+    except ValueError:
+        # There was a data type error which has already been
+        # logged.
+        return None
+
+    # webroot_map is, uniquely, a dict, and the general-purpose
+    # configuration restoring logic is not able to correctly parse it
+    # from the serialized form.
+    if "webroot_map" in renewalparams:
+        setattr(config.namespace, "webroot_map", renewalparams["webroot_map"])
+
+    try:
+        domains = [le_util.enforce_domain_sanity(x) for x in
+                   renewal_candidate.names()]
+    except (UnicodeError, ValueError):
+        logger.warning("Renewal configuration file %s references a cert "
+                       "that mentions a domain name that we regarded as "
+                       "invalid. Skipping.", full_path)
+        return None
+
+    setattr(config.namespace, "domains", domains)
+    return renewal_candidate
+
+
+def _renewal_conf_files(config):
+    """Return /path/to/*.conf in the renewal conf directory"""
+    return glob.glob(os.path.join(config.renewal_configs_dir, "*.conf"))
+
+
+def renew(cli_config, unused_plugins):
+    """Renew previously-obtained certificates."""
+    cli_config = configuration.RenewerConfiguration(cli_config)
+    if cli_config.domains != []:
+        raise errors.Error("Currently, the renew verb is only capable of "
+                           "renewing all installed certificates that are due "
+                           "to be renewed; individual domains cannot be "
+                           "specified with this action. If you would like to "
+                           "renew specific certificates, use the certonly "
+                           "command. The renew verb may provide other options "
+                           "for selecting certificates to renew in the future.")
+    for renewal_file in _renewal_conf_files(cli_config):
+        if not renewal_file.endswith(".conf"):
+            continue
+        print("Processing " + renewal_file)
+        # XXX: does this succeed in making a fully independent config object
+        #      each time?
+        config = configuration.RenewerConfiguration(copy.deepcopy(cli_config))
+        config.noninteractive_mode = True
+
+        # Note that this modifies config (to add back the configuration
+        # elements from within the renewal configuration file).
+        try:
+            renewal_candidate = _reconstitute(renewal_file, config)
+        except Exception as e: # pylint: disable=broad-except
+            # reconstitute encountered an unanticipated problem.
+            logger.warning("Renewal configuration file %s produced an "
+                           "unexpected error: %s. Skipping.", renewal_file, e)
+            logger.info("Traceback was:\n%s", traceback.format_exc())
+            continue
+
+        if renewal_candidate is None:
```

Comment:
nit: Rather than using continue, we could just change the logic to `if renewal_candidate is not None` and put the next three lines under that `if` statement.